eclipse-scout-core/src/desktop/outline/SearchOutline.ts

```diff
@@ -186,4 +270,133 @@ export class SearchOutline extends Outline implements SearchOutlineModel {
       elementToFocus.select();
     }
   }
+
+  updateSearchStates() {
+    const searchStates = new Set<SearchState>();
+
+    for (const page of this.nodes as SearchPage[]) {
+      if (page.searchState) {
+        searchStates.add(page.searchState);
+      }
+    }
+
+    this.setSearchStates(searchStates);
+  }
+
+  setSearchStates(searchStates: Set<SearchState>) {
+    searchStates ||= new Set();
+
+    const searchStatesToAdd = new Set<SearchState>();
+    for (const searchState of searchStates) {
+      if (this._searchStates.has(searchState)) {
+        continue;
+      }
+      searchStatesToAdd.add(searchState);
+    }
+
+    const searchStatesToRemove = new Set<SearchState>();
+    for (const searchState of this._searchStates) {
+      if (searchStates.has(searchState)) {
+        continue;
+      }
+      searchStatesToRemove.add(searchState);
+    }
+
+    searchStatesToAdd.forEach(searchState => this._addSearchState(searchState));
+    searchStatesToRemove.forEach(searchState => this._removeSearchState(searchState));
+  }
+
+  protected _addSearchState(searchState: SearchState) {
+    if (!searchState) {
+      return;
+    }
+
+    if (this._searchStates.has(searchState)) {
+      return;
+    }
+    this._searchStates.add(searchState);
+
+    this._installSearchStateListeners(searchState);
+    this._updateSearchStatus();
+  }
+
+  protected _removeSearchState(searchState: SearchState) {
+    if (!searchState) {
+      return;
+    }
+
+    if (!this._searchStates.has(searchState)) {
+      return;
+    }
+    this._searchStates.delete(searchState);
+
+    this._uninstallSearchStateListeners(searchState);
+    this._updateSearchStatus();
+  }
+
+  protected _installSearchStateListeners(searchState: SearchState) {
+    if (!searchState) {
+      return;
+    }
+    searchState.on('propertyChange:resultCount', this._searchStateResultCountChangedHandler);
+    searchState.on('propertyChange:limited', this._searchStateLimitedChangedHandler);
+    searchState.on('propertyChange:pending', this._searchStatePendingChangedHandler);
+    searchState.one('destroy', this._searchStateDestroyHandler);
+  }
+
+  protected _uninstallSearchStateListeners(searchState: SearchState) {
+    if (!searchState) {
+      return;
+    }
+    searchState.off('propertyChange:resultCount', this._searchStateResultCountChangedHandler);
+    searchState.off('propertyChange:limited', this._searchStateLimitedChangedHandler);
+    searchState.off('propertyChange:pending', this._searchStatePendingChangedHandler);
+    searchState.off('destroy', this._searchStateDestroyHandler);
+  }
+
+  protected _onSearchStateResultCountChanged(event: PropertyChangeEvent<number>) {
```

Comment:
Maybe just use one handler onSearchStateChange instead of 3 because they all do the same?